libraries/botbuilder/src/teamsInfo.ts

```diff
@@ -155,13 +158,33 @@ export class TeamsInfo {
             },
             activity: activity,
         } as ConversationParameters;
-        const connectorClient = (context.adapter as BotFrameworkAdapter).createConnectorClient(
-            context.activity.serviceUrl
-        );
-        const conversationResourceResponse = await connectorClient.conversations.createConversation(convoParams);
-        const conversationReference = TurnContext.getConversationReference(context.activity);
-        conversationReference.conversation.id = conversationResourceResponse.id;
-        return [conversationReference as ConversationReference, conversationResourceResponse.activityId];
+
+        let conversationReference: Partial<ConversationReference>;
+        let newActivityId: string;
+
+        if (botAppId) {
```

Comment:
This is a little misleading, if a `BotFrameworkAdapter` user passes in the `botAppId` parameter, they'll fall into this flow where the non-overridden `BotAdapter.createConversationAsync()` will [`throw new Error('NotImplemented');`](https://github.com/microsoft/botbuilder-js/blob/24a7d5fa62340eea8ec2304f0a6b25734af78787/libraries/botbuilder-core/src/botAdapter.ts#L190).

With some refactoring and a `try-catch` where if `'NotImplemented'` is the error message, default to the `BotFrameworkAdapter` logic used in the `else` block.

The docstring for the method should also be updated, as `botAppId` is only used in one scenario and should be explicitly communicated for a better developer experience.

Reply:
@stevengum How's [this](https://github.com/microsoft/botbuilder-js/pull/3821/commits/397bf87f6f78a4c2ae751eeb3585b61d48445797)? Is `if (botAppId && context.adapter instanceof CloudAdapter)` sufficient? I could implement the `try/catch`, it just wouldn't look nearly as clean.